src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java

```diff
@@ -217,13 +218,25 @@ public AppiumServiceBuilder withArgument(ServerArgument argument, String value)
             case "-g":
                 withLogFile(new File(value));
                 break;
+            case "--base-path":
+                serverArguments.put(argName,sanitizeBasePath(value));
```

Comment:
```suggestion
                serverArguments.put(argName, sanitizeBasePath(value));
```

Reply:
Noted on this, take care in the latest commit.